### PR TITLE
adjust versions

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes #<issue_number>

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: requirements/adjust_versions.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    # drop all except semantic version
+    ver = re.search(r'([\.\d]+)', ver).groups()[0]
+    # find candidates, by starting version pattern
+    options = [v for v in versions_all if v.startswith(ver)]
+    assert options, f"missing {ver} among {versions_all}"
+    # take the last one...
+    return sorted(options)[-1]
+
+
+def main(path_req: str, torch_version: str = None) -> None:
```
Comment (kaushikb11) [inline requirements/adjust_versions.py:27]:
```suggestion
def main(path_req: str, torch_version: Optional[str] = None) -> None:
```

Comment (kaushikb11) [reply requirements/adjust_versions.py:27]: Nit

**Review by kaushikb11 — APPROVED** [review]

File: requirements/adjust_versions.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import os
+import re
+import sys
+from typing import Any, Dict
+
+VERSIONS_LUT: Dict[str, Dict[str, Any]] = {
```
Comment (carmocca) [inline requirements/adjust_versions.py:6]:
```suggestion
VERSIONS_LUT: Dict[str, Dict[str, str]] = {
```

**Review by carmocca — APPROVED** [review]

**Comment by carmocca** [comment]
whoops, didn't realise auto-merge was enabled and I was the missing reviewer :/

**Pull request merged**